File: _2020/go/days/day1.go
package days

import (
	"fmt"
	"strconv"
	"strings"

	"advent/utils"
)

func parseInput() []int {
	data := utils.LoadPuzzleInput("day1")
	input_strings := strings.Fields(data)
	integers := make([]int, len(input_strings))
	for ind, str := range input_strings {
		i, _ := strconv.Atoi(str)
		integers[ind] = i
	}
	return integers
}

func Day1Part1() string {
	integers := parseInput()
	value := 0
	for _, a := range integers {
		for _, b := range integers {
			if a+b == 2020 {
				value = a * b
				break
			}
		}
	}
	return fmt.Sprintf("%v", value)
}

func Day1Part2() string {
	integers := parseInput()
	value := 0
	for _, a := range integers {
		for _, b := range integers {
			for _, c := range integers {
				if a+b+c == 2020 {
					value = a * b * c
					break
				}
			}
		}
	}
	return fmt.Sprintf("%v", value)
}

func Day1() {
	utils.Profile(Day1Part1)
	utils.Profile(Day1Part2)
}

File: typescript/_2020/day1.ts
import {loadPuzzleInput, profile} from "../utils";

function parseInput(): number[] {
    const input = loadPuzzleInput("2020/day1")
    return input.split("\n").map(s => parseInt(s, 10))
}

export function day1Part1(): number {
    const integers = parseInput()
    for (let a of integers) {
        for (let b of integers) {
            if (a + b === 2020) {
                return a * b
            }
        }
    }
}

export function day1Part2(): number {
    const integers = parseInput()
    for (let a of integers) {
        for (let b of integers) {
            for (let c of integers) {
                if (a + b + c === 2020) {
                    return a * b * c
                }
            }
        }
    }
}

if (require.main === module) {
    profile(day1Part1)
    profile(day1Part2)
}